### PR TITLE
Energy dashboard: Document required parameter and references to SML and image recognition data sources to energy documentation

### PR DESCRIPTION
**Pull request by nohn**
## Proposed change

Looking at the home assistant community, stack overflow etc., it seems quite an issue for many people to manually integrate their energy sensors. Adding information about the required paramters and examples likely would make things easier for those.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/energy/electricity-grid.markdown [hunk 1]
```diff
@@ -59,6 +63,38 @@ _Attention! Installing CT clamp sensor devices requires opening your electrical
 
 Some energy providers will provide you real-time information about your usage and have this data integrated into Home Assistant.
 
+### Manual Integration
```
Comment (frenck) [inline source/_docs/energy/electricity-grid.markdown:72]: This page provides general information on energy integration. The example provided below is very specific for MQTT which IMHO doesn't belong on this page.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_docs/energy/water.markdown [hunk 2]
```diff
@@ -26,4 +26,23 @@ Alternatively, the following shops sell ESPHome-based devices, that use a proxim
 
 Maybe you like to build one yourself? Pieter Brinkman has quite a [nice blog article on how to create your own water sensor](https://www.pieterbrinkman.com/2022/02/02/build-a-cheap-water-usage-sensor-using-esphome-home-assistant-and-a-proximity-sensor/) using ESPHome, or [build a water meter](https://www.ztatz.nl/p1-monitor-watermeter/) that works with the [P1 Monitor](/integrations/p1_monitor) integration.
 
+To make your water usage information available to the energy dashboard, you need to provide `device_class`, `state_class` and `unit_of_measurement`. 
+For example:
```
Comment (frenck) [inline source/_docs/energy/water.markdown]: Same comment as above

File: source/_docs/energy/electricity-grid.markdown (hunk 1 above)
Comment (nohn) [reply source/_docs/energy/electricity-grid.markdown:72]: Example removed.

File: source/_docs/energy/water.markdown (hunk 2 above)
Comment (nohn) [reply source/_docs/energy/water.markdown]: Example removed.

File: source/_docs/energy/electricity-grid.markdown [hunk 3]
```diff
@@ -65,35 +65,7 @@ Some energy providers will provide you real-time information about your usage an
...
-    device_class: energy
-    name: 'Mains Total Energy Outbound'
-    state_class: total_increasing
-    unique_id: 'mains_total_outbound_energy'
-    unit_of_measurement: 'kWh'
-```
-
-{% endraw %}
-
+If you manually integrate your sensor, `device_class`, `state_class` and `unit_of_measurement` have to be provided.
```
Comment (frenck) [inline source/_docs/energy/electricity-grid.markdown]: Let's make more clear what manual means. How about something like this?

```suggestion
If you manually integrate your sensors, for example, using the [MQTT](/integrations/mqtt) or [Template](/integrations/template) integrations: Make sure you set and provide the `device_class`, `state_class`, and `unit_of_measurement` for those sensors.
```

**Comment by nohn** [comment]
@frenck are there still any unresolved issues? Github says so, but I can find them in the code reviews.

**Review by frenck — CHANGES_REQUESTED** [review]
I'm sorry for letting this hang for a bit @nohn 🙏 

In the mean time, a merge conflict occurred, could you take a look?

Thanks! 👍 

../Frenck

File: source/_docs/energy/water.markdown [hunk 4]
```diff
@@ -31,4 +31,11 @@ Maybe you like to build one yourself?
 ![Photo of the AI-on-the-edge-device Workflow](/images/docs/energy/ai-on-the-edge-device.jpg)
 
 
+Another common alternative is using image recognition using one of the following.
+
+- [AI-on-the-edge-device](https://github.com/jomjol/AI-on-the-edge-device) running machine learning algorithms on an ESP32
```
Comment (frenck) [inline source/_docs/energy/water.markdown]: This one is listed already directly above.

```suggestion
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — APPROVED** [review]
Awesome, Thanks, @nohn 👍 

../Frenck

**Pull request merged**